psh/ifconfig/ifconfig.c

```diff
@@ -0,0 +1,503 @@
+		case IFF_BROADCAST:
+			return "BROADCAST";
+		case IFF_DEBUG:
+			return "DEBUG";
+		case IFF_LOOPBACK:
+			return "LOOPBACK";
+		case IFF_POINTOPOINT:
+			return "POINTOPOINT";
+		case IFF_NOTRAILERS:
+			return "NOTRAILERS";
+		case IFF_RUNNING:
+			return "RUNNING";
+		case IFF_NOARP:
+			return "NOARP";
+		case IFF_PROMISC:
+			return "PROMISC";
+		case IFF_ALLMULTI:
+			return "ALLMULTI";
+		case IFF_OACTIVE:
+			return "OACTIVE";
+		case IFF_SIMPLEX:
+			return "SIMPLEX";
+		case IFF_LINK0:
+			return "LINK0";
+		case IFF_LINK1:
+			return "LINK1";
+		case IFF_MULTICAST:
+			return "MULTICAST";
+		case IFF_DYNAMIC:
+			return "DYNAMIC";
+		default:
+			return "UNKNOWN_FLAG";
+	}
+}
+
+
+static inline int psh_ifconfigPrintInterface(const struct ifaddrs *interface, int sd)
+{
+	unsigned int flags, i;
+	struct in_addr addr, broadcast, mask;
+	int ret;
+	struct ifreq ioctlInterface;
+
+	(void)strncpy(ioctlInterface.ifr_name, interface->ifa_name, IFNAMSIZ - 1);
+	/* Get interface flags */
+	ret = ioctl(sd, SIOCGIFFLAGS, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFFLAGS)");
+		return ret;
+	}
+	flags = ioctlInterface.ifr_flags;
+
+	/* Get interface IP address */
+	ret = ioctl(sd, SIOCGIFADDR, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFADDR)");
+		return ret;
+	}
+	addr = ((struct sockaddr_in *)&ioctlInterface.ifr_addr)->sin_addr;
+
+	/* Get interface broadcast IP address */
+	ret = ioctl(sd, SIOCGIFBRDADDR, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFBRDADDR)");
+		return ret;
+	}
+	broadcast = ((struct sockaddr_in *)&ioctlInterface.ifr_addr)->sin_addr;
+
+	/* Get interface IP mask */
+	ret = ioctl(sd, SIOCGIFNETMASK, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFNETMASK)");
+		return ret;
+	}
+	mask = ((struct sockaddr_in *)&ioctlInterface.ifr_addr)->sin_addr;
+
+	printf("%-10s", interface->ifa_name);
+
+	if ((flags & IFF_LOOPBACK) == 0) {
+		/* Get interface hardware address */
+		(void)memset(&ioctlInterface.ifr_hwaddr, 0, sizeof(struct sockaddr));
+		ret = ioctl(sd, SIOCGIFHWADDR, &ioctlInterface);
+		if (ret < 0) {
+			perror("ioctl(SIOCGIFHWADDR)");
+			return ret;
+		}
+		printf(" HWAddr");
+		for (ret = 0; ret < 6; ++ret) {
+			printf(":%02hhx", ioctlInterface.ifr_hwaddr.sa_data[ret]);
+		}
+	}
+
+	printf("\n%10sinet addr:%s ", "", inet_ntoa(addr));
+	printf("Broadcast:%s ", inet_ntoa(broadcast));
+	printf("Mask:%s\n", inet_ntoa(mask));
+	if (flags != 0) {
+		printf("%10s", "");
+		for (i = 1; i <= flags; i <<= 1) {
+			if ((flags & i) != 0) {
+				printf("%s ", psh_ifconfigPrintFlag(i));
+			}
+		}
+	}
+
+	/* Get MTU */
+	ret = ioctl(sd, SIOCGIFMTU, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFMTU)");
+		return ret;
+	}
+	printf("MTU:%d", ioctlInterface.ifr_mtu);
+	/* Get metric */
+	ret = ioctl(sd, SIOCGIFMETRIC, &ioctlInterface);
+	if (ret < 0) {
+		perror("ioctl(SIOCGIFMETRIC)");
+		return ret;
+	}
+	printf(" Metric:%d", ioctlInterface.ifr_metric);
```

Comment:
Also found strange behaviour comparing to busybox ifconfig. We need to check if (our or busybox) ifconfig show correct values of METRIC
![image](https://github.com/phoenix-rtos/phoenix-rtos-utils/assets/49757239/5ee3b514-df96-416b-a5b6-3b9cdbb07b13)

Reply:
That is weird. I don't know why there is a difference.

Reply:
In busybox at `./_build/ia32-generic/busybox/busybox-1.27.2/networking/interface.c:1020`	we have line `printf(" MTU:%d  Metric:%d", ptr->mtu, ptr->metric ? ptr->metric : 1);`.

As you can see, it displays 1 if metric is 0.

Reply:
Ok, lets replicate this behaviour